### ISSUE TITLE
Python: Replace 'str' with 'string_types' in generated types (for python 2)

### ISSUE THREAD

**Issue by lucasvanhalst** [Jumpscale/go-raml#313]
Currently the generated code for types looks like this: 

```python
class TestType(object):
    """
    auto-generated. don't touch.
    """
    def __init__(self, json=None, **kwargs):
        if not json and not kwargs:
            raise ValueError('No data or kwargs present')

        class_name = 'TestType'
        create_error = '{cls}: unable to create {prop} from value: {val}: {err}'
        required_error = '{cls}: missing required property {prop}'

        data = json or kwargs

        property_name = 'testprop'
        val = data.get(property_name)
        if val is not None:
            datatypes = [str]  # < --- this is a problem for python 2.7
            try:
                self.testprop = client_support.val_factory(val, datatypes)
            except ValueError as err:
                raise ValueError(create_error.format(cls=class_name, prop=property_name, val=val, err=err))
        else:
            raise ValueError(required_error.format(cls=class_name, prop=property_name))
```

Inside the `if val is not None` statement, datatypes is set to `[str]`. Since in python2.7 there is both 'str' and 'unicode' for string types, it errors because a unicode string is not an instance of `str`.

Replace this with 'string_types' from the 'six' library as this includes both:
```python
from six import string_types
>>> a = 'test'
>>> b = u'test'
>>> isinstance(a, str)
True
>>> isinstance(b, str)
False
>>> isinstance(a, unicode)
False
>>> isinstance(b, unicode)
True
>>> isinstance(a, string_types)
True
>>> isinstance(b, string_types)
True
```

If adding six is a bit too much, use this:

```python
import sys
PY3 = sys.version_info[0] == 3

if PY3:
    string_types = str
else:
    string_types = basestring
```

**Comment by iwanbk** [comment]
How important is this feature?
Because we indeed only support python 3?

**Comment by lucasvanhalst** [comment]
Well we use Google appengine standard and only python 2 is supported there, so this would be nice. There's a couple other incompatibilities but those are easier to fix manually

**Comment by iwanbk** [comment]
blocked by #344 or will be done at that issue

**Comment by iwanbk** [comment]
fixed at #351
